sdk/Lusid.Sdk/Utilities/ApiRetryHandler.cs

```diff
@@ -0,0 +1,68 @@
+﻿using System;
+using Polly;
+using RestSharp;
+
+namespace Lusid.Sdk.Utilities
+{
+    /// <summary>
+    /// Class used to define API error retry rules for all API calls.
+    /// </summary>
+    public static class ApiRetryHandler
+    {
+        /// <summary>
+        /// Number of max retry attempts
+        /// </summary>
+        public const int MaxRetryAttempts = 3;
+
+        /// <summary>
+        /// Get the internal exception condition on which to retry.
+        /// </summary>
+        /// <param name="restResponse">Response object that comes from the API Client</param>
+        /// <returns></returns>
+        private static bool GetInternalExceptionRetryCondition(IRestResponse restResponse)
+        {
+            return restResponse.ErrorException != null || restResponse.StatusCode == 0;
+        }
+
+        /// <summary>
+        /// Define Polly retry policy for synchronous API calls. Handles internal SDK exceptions only.
+        /// Use .Wrap() method to combine this policy with your other custom policies
+        /// </summary>
+        public static Policy<IRestResponse> GetInternalExceptionRetryPolicy() =>
+            Policy
+                .Handle<SystemException>()
+                .OrResult<IRestResponse>(GetInternalExceptionRetryCondition)
+                .Retry(
+                    MaxRetryAttempts,
+                    onRetry: (result, retryCount, context) =>
+                    {
+                        Console.WriteLine("An internal exception has occurred. Retrying. " +
+                                          $"Retry attempt: {retryCount}. Max retry attempts: {MaxRetryAttempts}");
+
+                        Console.WriteLine(
+                            "Temporarily logging the exception stack trace for finding the underlying root cause of this exception:\n" +
+                            $"{result.Result.ErrorException}\n");
+                    });
```

Comment:
Why is this temporary?

Reply:
Because when we get these errors, we want to at least know exactly where they originate, so that if we actually find a way to fix it in the future, we know where to start

Reply:
This is a production SDK I don't think we should temporarily write anything to the console. If you want to expose this information we should find a way to allow the user to decide what log level to expose. Here this will be printed whether you want it or not.

Reply:
We don't have a logging framework implemented, so will this suffice?

```
        private static void HandleRetryAction(DelegateResult<IRestResponse> result, int retryCount, Context context)
        {
            if (Environment.GetEnvironmentVariable("HIDE_INTERNAL_EXCEPTION_RETRY_LOGS") == "true") return;
            
            Console.WriteLine("An internal exception has occurred. Retrying. " +
                              $"Retry attempt: {retryCount}. Max retry attempts: {MaxRetryAttempts}");

            Console.WriteLine(
                "Temporarily logging the exception stack trace for finding the underlying root cause of this exception:\n" +
                $"{result.Result.ErrorException}\n");
        }
```

Reply:
No that's okay, let's leave it in and we can address the logging in a separate merge request.